packages/components/src/components/assets/AssetShareIndicator/AssetShareIndicator.tsx

```diff
@@ -10,25 +10,109 @@ const Container = styled.div`
     align-items: center;
     display: flex;
     justify-content: center;
-    width: 48px;
-    height: 48px;
+    width: ${spacingsPx.xxxxl};
+    height: ${spacingsPx.xxxxl};
     border-radius: 50%;
 `;
 
+const rgbToHexColor = (rgbColor: string) => {
+    const [r, g, b] = rgbColor.split('-').map(Number);
+    const hex = [r, g, b].map(x => x.toString(16).padStart(2, '0')).join('');
+
+    return '#' + hex;
+};
+
+const getDominantColor = (imageData: ImageData) => {
+    const colorMap: { [key: string]: number } = {};
+    let dominantKey = '';
+    let maxCount = 0;
+
+    for (let i = 0; i < imageData.data.length; i += 4) {
+        const alpha = imageData.data[i + 3];
+        if (alpha !== 255) {
+            continue; // ignore pixel with transparency
+        }
+
+        const key = `${imageData.data[i]}-${imageData.data[i + 1]}-${imageData.data[i + 2]}`;
```

Comment:
optional/opinionated: I would suggest to be a bit more specific `key` -> `imagePixelKey`?